cmd/ci-operator-configresolver/main.go

```diff
@@ -158,20 +161,101 @@ func resolveAndRespond(registryAgent agents.RegistryAgent, config api.ReleaseBui
 		logger.WithError(err).Warning("failed to resolve config with registry")
 		return
 	}
-	jsonConfig, err := json.MarshalIndent(config, "", "  ")
+	writeJSON(w, logger, config)
+}
+
+func writeJSON(w http.ResponseWriter, logger *logrus.Entry, data interface{}) {
+	jsonData, err := json.MarshalIndent(data, "", "  ")
 	if err != nil {
-		metrics.RecordError("failed to marshal config", configresolverMetrics.ErrorRate)
+		metrics.RecordError("failed to marshal JSON", configresolverMetrics.ErrorRate)
 		w.WriteHeader(http.StatusInternalServerError)
-		fmt.Fprintf(w, "failed to marshal config to JSON: %v", err)
-		logger.WithError(err).Errorf("failed to marshal config to JSON")
+		fmt.Fprintf(w, "failed to marshal JSON: %v", err)
+		logger.WithError(err).Errorf("failed to marshal JSON")
 		return
 	}
+	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
-	if _, err := w.Write(jsonConfig); err != nil {
+	if _, err := w.Write(jsonData); err != nil {
 		logrus.WithError(err).Error("Failed to write response")
 	}
 }
 
+func writeErrorPage(w http.ResponseWriter, pageErr error, status int) {
+	w.Header().Set("Content-Type", "text/plain;charset=UTF-8")
+	w.WriteHeader(status)
+	fmt.Fprintf(w, "%v\n", pageErr)
+}
+
+func resolveStepRegistry(regAgent agents.RegistryAgent, confAgent agents.ConfigAgent) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		trimmedPath := strings.TrimPrefix(req.URL.Path, req.URL.Host)
+		// remove leading path prefix
+		trimmedPath = strings.TrimPrefix(trimmedPath, "/step-registry/")
+		// remove trailing slash
+		trimmedPath = strings.TrimSuffix(trimmedPath, "/")
+		splitURI := strings.Split(trimmedPath, "/")
+		if len(splitURI) == 1 {
```

Comment:
if we did e.g. `GET /step/{name}` and the like for `chain` this could be simpler

Reply:
this is just copy/pasted out of pkg/webreg.  If the implementation there improves in some future PR, I can copy/paste the improved logic in here to pick it up.

Reply:
but these are separate APIs doing separate things. there will never be a search page here ...

Reply:
Folks won't want to search for jobs and such?